tests/Unit/NumericalAlgorithms/Spectral/Test_Spectral.cpp

```diff
@@ -157,28 +241,91 @@ SPECTRE_TEST_CASE("Unit.Numerical.Spectral.ExactQuadrature",
+                          Spectral::Quadrature::Gauss>(4, 7, 35. * M_PI / 16.);
+    test_exact_quadrature<Spectral::Basis::Chebyshev,
+                          Spectral::Quadrature::Gauss>(5, 9,
+                                                       315. * M_PI / 128.);
+    test_exact_quadrature<Spectral::Basis::Chebyshev,
+                          Spectral::Quadrature::Gauss>(6, 11,
+                                                       693. * M_PI / 256.);
+  }
+  SECTION(
+      "Chebyshev-Gauss-Lobatto quadrature is exact to polynomial order "
+      "2*num_points-3") {
+    test_exact_quadrature<Spectral::Basis::Chebyshev,
+                          Spectral::Quadrature::Gauss>(2, 1, M_PI);
+    test_exact_quadrature<Spectral::Basis::Chebyshev,
+                          Spectral::Quadrature::Gauss>(3, 3, 3. * M_PI / 2.);
+    test_exact_quadrature<Spectral::Basis::Chebyshev,
+                          Spectral::Quadrature::Gauss>(4, 5, 15. * M_PI / 8.);
+    test_exact_quadrature<Spectral::Basis::Chebyshev,
+                          Spectral::Quadrature::Gauss>(5, 7, 35. * M_PI / 16.);
+    test_exact_quadrature<Spectral::Basis::Chebyshev,
+                          Spectral::Quadrature::Gauss>(6, 9,
+                                                       315. * M_PI / 128.);
+  }
+}
+
+namespace {
+
+template <Spectral::Basis BasisType, Spectral::Quadrature QuadratureType>
+void test_exact_integral(std::function<size_t(size_t)> max_poly_deg) {
+  for (size_t n = Spectral::minimum_number_of_points<BasisType, QuadratureType>;
+       n <= Spectral::maximum_number_of_points<BasisType>; n++) {
+    for (size_t p = 0; p <= max_poly_deg(n); p++) {
+      const auto collocation_pts =
+          Spectral::collocation_points<BasisType, QuadratureType>(n);
+      const auto weights =
+          Spectral::integral_weights<BasisType, QuadratureType>(n);
+      const DataVector u = polynomial(p, collocation_pts);
+      const double numeric_integral = ddot_(n, u.data(), 1, weights.data(), 1);
+      const double analytic_integral = polynomial_integral(p);
+      CHECK_ITERABLE_APPROX(analytic_integral, numeric_integral);
     }
   }
 }
+
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Numerical.Spectral.ExactIntegral",
+                  "[NumericalAlgorithms][Spectral][Unit]") {
+  // Legendre integrals are identical to quadrature since they have unit weight.
+  SECTION(
+      "Legendre-Gauss integral is exact to polynomial order 2*num_points-1") {
+    test_exact_integral<Spectral::Basis::Legendre, Spectral::Quadrature::Gauss>(
+        [](const size_t n) { return 2 * n - 1; });
+  }
+  SECTION(
+      "Legendre-Gauss-Lobatto integral is exact to polynomial order "
+      "2*num_points-3") {
+    test_exact_integral<Spectral::Basis::Legendre,
+                        Spectral::Quadrature::GaussLobatto>(
+        [](const size_t n) { return 2 * n - 3; });
+  }
+  // Is there a bound on the error of Chebyshev integrals that we can test?
```

Comment:
Can anyone point me to literature on this? I'm only finding bounds on $\int f(x)w(x)dx$...